pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -95,6 +96,9 @@ public abstract class AbstractTopic implements Topic, TopicPolicyListener {
     protected static final long POLICY_UPDATE_FAILURE_RETRY_TIME_SECONDS = 60;
 
     protected final String topic;
+    @Getter
+    @Setter
+    protected volatile CompletableFuture<Optional<Topic>> createFuture;
```

Comment:
It would be useful to document the purpose of this `createFuture`. Without documentation, it could be hard to understand the reason why this field exists.
Please add javadocs. It could be useful to use ordinary getter and setter methods so that javadocs could be added.

Reply:
I wonder if it would make more sense to call this field `cacheKey`?

Reply:
Thanks for reviewing and adding documentation. I believe keeping the name as createFuture makes it clearer.